hudi-hadoop-common/src/test/java/org/apache/hudi/common/table/read/TestCustomMerger.java

```diff
@@ -182,6 +180,7 @@ public void testWithThreeLogFiles(boolean useRecordPositions) throws IOException
 
   @Test
   public void testWithFourLogFiles() throws IOException, InterruptedException {
+    shouldWritePositions = Arrays.asList(false, false, false, false, false);
```

Comment:
Should this be parameterized on `boolean useRecordPositions` as other tests with     `shouldWritePositions = Arrays.asList(useRecordPositions, useRecordPositions, useRecordPositions, useRecordPositions, useRecordPositions)`?

Reply:
I don't think we're getting much out of all of these combinations. The unit testing needs to be redone here so that we focus on the unit under test instead of a wider range of things

Reply:
The positional based lookup and fallback logic is not at all related to the CustomMerger so these tests need to be moved to a class that just tests the record buffer lookup/fallback